### DISCUSSION TITLE
API 호출 에러 핸들링 어디까지..?!!

### DISCUSSION THREAD

**Discussion by kkojae91** [woowacourse-study/2022-code-review-study#55] [Youtube] [answered]
API 호출시 에러가 발생할 경우 에러 핸들링을 각각 해주는게 좋은 것 같은데... 어디까지 해줘야 할까요??! (물론 다 해줘야겠지만, mdn status 코드를 확인해보니 엄청 많더라구요!)

[#140 리뷰어 코맨트](https://github.com/woowacourse/javascript-youtube-classroom/pull/140#discussion_r830565603)
[mdn-status-code](https://developer.mozilla.org/ko/docs/Web/HTTP/Status)

아래는 제가 담당한 크루원들이 특정 error에 대해 후 처리를 해준 코드를 긁어 왔습니다.

#### failed to fetch

```js
const handleError = (errorMessage) => {
  switch (errorMessage) {
    case "Failed to fetch":
      alert("인터넷 연결이 원활하지 않습니다. 잠시 후 다시 시도해주세요.");
      break;
    case "":
      break;
  }
};
```

#### 403 error

```js
function checkErrorCodeNumber(errorCode) {
  if (errorCode === 403) {
    alert(ERROR_MESSAGE.API_QUOTA_EXCEEDED);
  }
}
```

```js
callSearchAPI(searchKeyword) {
  const URL = this.#searchResults
    ? this.getURL(searchKeyword, this.#searchResults.nextPageToken)
    : this.getURL(searchKeyword);
  return fetch(URL)
    .then(response => {
      if (response.ok) {
        return response.json();
      }
      if (response.status === ERROR.CODE.API_DISABLE) {
        throw new Error(ERROR.MESSAGE.API_DISABLE);
      }
    })
    .catch(error => alert(error));
}
```

**Comment by byhhh2** [comment, marked as answer]
이것도 정말 어렵네요. 저는 `사용자가 alert을 통해 해결할 수 있냐!`에 중점을 둘 것 같아요.. 사용자가 해결할 수 있는 부분을 중심적으로 예외처리를 하는 것이죠,,

만약에 API 한도가 넘었다!, 서버가 터졌다. =>  사용자가 해결 불가, 통틀어서 에러 처리를 해준다. (사용자에게는 잠시 후에 시도하라고 말해준다던지)
사용자가 이상한 키워드를 사용해서 검색했다, 네트워크 상태가 별로다 =>  사용자가 해결 가능, --를 해달라고 직접적으로 얘기한다.

그런데 이런 기준을 정해도 어떤 케이스가 어떤 status code가 나오는지 .. 알아봐야겠네요.. (혼란)

**Reply by wonsss** [reply]
5xx -> 백엔드한테 욕하기
4-> 내 잘못
2-> 아무튼 맞음

**Reply by wonsss** [reply]
"이건 서버개발자가 착해야해요
까다로운 분들은 안내려줌
무조건 성공(데이터 페치하면 무조건 성공)으로 처리하고 지만의 에러 커스텀 상태 주는 경력자 엄청 많아요
그러면 프론트개발자가 해야죠. 서버개발자 : "프론트야 난 모르겠고 니가해""

따라서, 서버개발자랑 협업을 잘 하자..!

**Reply by lokba** [reply]
swagger api 명세서
